src/tls_stream.rs

```diff
@@ -79,6 +79,17 @@ impl Builder {
         self
     }
 
+    ///DEPRECATED this connect will not return a HandshakeError::Failure2
+    #[deprecated(note = "please use connect2 instead")]
+    pub fn connect<S>(&mut self,
+                      cred: SchannelCred,
+                      stream: S)
+                      -> Result<TlsStream<S>, HandshakeError<S>>
+        where S: Read + Write
+    {
+        self.initialize(cred, false, stream)
```

Comment:
Can we just have this delegate to connect2 and adjust the error type to avoid having to duplicate a bunch of stuff?